lib/parallel.rb

```diff
@@ -485,7 +524,8 @@ def process_incoming_jobs(read, write, job_factory, options, &block)
     end
 
     def handle_exception(exception, results)
-      return nil if [Parallel::Break, Parallel::Kill].include? exception.class
+      return if [Parallel::Break, Parallel::Kill].include?(exception.class)
+      return exception if exception.class == Parallel::Return
```

Comment:
I'd expect this to return the value ?
```suggestion
      return exception.result if exception.class == Parallel::Return
```